### DISCUSSION TITLE
[Question][Hw6] Humble proposal for Assignment 6

### DISCUSSION THREAD

**Discussion by AmabileDelicato** [prosyslab-classroom/cs348-information-security#633] [Q&A]
Dear instructors

First, I'd like to say thank you for providing a really fine, greate lectures and assistance for us this semester.
We've just informed about the last assignment and thankfully, through this assignment, we hope to enhance and verify what we have learned so far in the lecture.

Also we could find that it is in several T/F questions and narrative ones, and those questions are really fine that they can facilitate students improve their own learning.

Some might be concerned, however, because of the format of the assignment. Compare to the previous assignments that is consisted of programming works and partial T/F questions, assignment 6 is composed with totally T/F and some prose questions.

Asking questions to students have the advantage of being able to easily grasp the students' understanding, but also has a disadvantage that the burden of getting the problem wrong becomes strictly heavier.

Since programming assignments is gradual, we could expect that we can earn points for our efforts almost linearly.
But in questions, getting only single one of them wrong will cause significant amount of degraded score, like -10, -15, and such a score is the amount of that when making a very HUGE mistakes in programming tasks.

Also questions might be vulnerable to cheating, plagiarism, since some crews can cooperate and share their ideas and write the answer. Even if some plagiarism detectors try to investigate such malicious behavior, it is likely to fail since the key point of the answer in here is one key idea, not the literary/writing style. That way they can bypass the suspicion by simply writing the sentence in a different way while keeping their shared idea.

**For these reasons, I respectfully propose the way of slightly reducing the weight of this assignment compared to the previous assignments.**

We know that all the grading weight/criteria is reasonable, and there's a good meaning by the TA beyond such decision.
I'm sorry if this proposal has caused you any inconvenience.

Thank you for reading this humble proposal.

Again, thank you for giving us nice lectures and being great facilitators which provided us opportunities to grow ourselves.

Best regards.

**Comment by minsukk-at-kaist-ac-kr** [comment]
Dear @AmabileDelicato,

Thanks for your thoughtful suggestion for Assignment 6. I hear your concern. We will try to offer partial scores as much as possible when grading answers so that students won’t be heavily penalized for not knowing the right answer. 

Re: weight adjustment for Assignment 6. As you’d all agree, fairness is important. The equal weights for all assignments was announced in our first lecture and every one has been allocated his/her resources throughout the semester according to this rule. Changing such a very critical rule of the game at the end of the semester is never considered right. Our teaching crew will do our best to produce a fair evaluation of all the assignments. Kindly do let us know whenever you feel you are unfairly evaluated for your submissions. 

Last, please understand that the course is designed to have no exams but instead have six assignments. We, the two instructors, believe that conducting midterm/final exams in remote settings would impose unnecessary pressure and burden to our students, and hence we decided to administer six assignments. 

Feel free to comment or email me if you have any follow-up questions.

Cheers,
Min Suk
